libyul/AsmParser.cpp

```diff
@@ -130,37 +130,73 @@ void Parser::fetchSourceLocationFromComment()
 	if (m_scanner->currentCommentLiteral().empty())
 		return;
 
-	static regex const lineRE = std::regex(
-		R"~~~((^|\s*)@src\s+(-1|\d+):(-1|\d+):(-1|\d+)(\s+|$))~~~",
-		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	static regex const tagRegex = regex(
+		R"~~(\s*(@[a-zA-Z0-9\-_]+)\s+)~~",             // tag, e.g: @src
```

Comment:
What prevents this from matching `a@src`? There should be a space in front of the `@`.

Reply:
Nothing. It was accepted as a valid tag before my changes and the PR does not change that. I added a test case to show this.

Should I change it to require whitespace before?

Reply:
Changed to require whitespace.